### PR TITLE
[WIP] Replace addField HOC with useField hook

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: packages/ra-core/src/form/FormField.tsx [hunk 1]
```diff
@@ -21,20 +22,11 @@ interface Props extends Omit<FieldProps<any, HTMLElement>, 'validate'> {
 }
 
 export const FormField: SFC<Props> = ({ input, validate, ...props }) => {
```
Comment (fzaninotto) [inline packages/ra-core/src/form/FormField.tsx]: input is injected by the Field, it should not be in the props, should it?

Comment (fzaninotto) [reply packages/ra-core/src/form/FormField.tsx]: Sorry, disregard my comment

Comment (djhi) [reply packages/ra-core/src/form/FormField.tsx]: It's here to check wether we need to inject it (for backward compatibility). If present, we don't need to inject the props from the hook

File: packages/ra-ui-materialui/src/input/DateTimeInput.tsx [hunk 2]
```diff
@@ -15,7 +16,7 @@ const leftPad2 = leftPad(2);
  * @returns {String} A standardized datetime (yyyy-MM-ddThh:mm), to be passed to an <input type="datetime-local" />
  */
 const convertDateToString = v => {
-    if (!(v instanceof Date) || isNaN(v)) return '';
+    if (!(v instanceof Date) || isNaN(v.getDate())) return '';
```
Comment (Kmaschta) [inline packages/ra-ui-materialui/src/input/DateTimeInput.tsx]: I'm confused about `v.getDate` here..

If `v` is a Date, this function returns the day of the month, which is always a number
If `v` is a date from moment or date-fns, it returns a Date, and is never a number

Am I missing something?

Also, the `convertDateToString` doesn't respect the ISO 8601 standard: it misses the timezone at the end (should be `Z` or `+00` something).

All thing considered, can we replacethis whole function by `date.toISOString()`, or do we need to strip the timezone?

Comment (djhi) [reply packages/ra-ui-materialui/src/input/DateTimeInput.tsx]: I haven't think about it, just trying to convert those inputs to TypeScript and make them use the hook from react-final-form, without changing too many things.

Comment (djhi) [reply packages/ra-ui-materialui/src/input/DateTimeInput.tsx]: If you think this is a prb, open an issue and we'll address it in a dedicated PR

**Comment by djhi** [comment]
Closing this in favor of multiple smaller PRs

**Pull request closed without merging**